src/main/java/Evelyn.java

```diff
@@ -0,0 +1,129 @@
+import java.util.Scanner;
+
+public class Evelyn {
+
+    public static Task[] tasks;
+    public static int indexOfTask = 0;
+
+    public static void markTask(int index, boolean done) {
+        if (index >= 0 && index < indexOfTask) {
+            if (done) {
+                tasks[index].markAsDone();
+                System.out.println("Nice! I've marked this task as done:");
+            } else {
+                tasks[index].markAsNotDone();
+                System.out.println("OK, I've marked this task as not done yet:");
+            }
+            System.out.println("  " + tasks[index].getStatusIcon() + " " + tasks[index].description);
+            printLine();
+        } else {
+            System.out.println("Invalid task index. Please try again.");
+            printLine();
+        }
+    }
+
+    public static void printList(Task[] tasks) {
+        int index = 1;
+        for (Task task : tasks) {
+            if (task == null) {
+                printLine();
+                break;
+            } else {
+                System.out.println(index + ". " + tasks[index - 1].toString());
+                index++;
+            }
+        }
+    }
+
+    public static void echo() {
+        String line;
+        System.out.println("type your command: ");
+        Scanner in = new Scanner(System.in);
+        line = in.nextLine().trim();
+        printLine();
+        if (line.equals("bye")) {
+            return;
+        } else if (line.equals("list")) {
+            System.out.println("Here are the tasks in your list:");
+            printList(tasks);
+        } else if (line.startsWith("mark")) {
+                int index = Integer.parseInt(line.substring(5).trim()) - 1;
+                markTask(index, true);
+        } else if (line.startsWith("unmark")) {
+            int index = Integer.parseInt(line.substring(7).trim()) - 1;
+            markTask(index, false);
+
+        } else if (line.startsWith("todo")) {
+            tasks[indexOfTask] = new Todos(line.substring(5).trim());
+            indexOfTask++;
+            System.out.println("Got it. I've added this task:");
+            System.out.println("  " + tasks[indexOfTask - 1]);
+            System.out.println("Now you have " + indexOfTask + " tasks in the list.");
+            printLine();
+        } else if (line.startsWith("deadline")) {
+            boolean haveBy = line.contains("/by");
+            if (haveBy) {
+                String[] parts = line.substring(9).split("/by");
+                String description = parts[0].trim();
+                String by = parts[1].trim();
+                tasks[indexOfTask] = new Deadlines(description, by);
+                indexOfTask++;
+                System.out.println("Got it. I've added this task:");
+                System.out.println("  " + tasks[indexOfTask - 1]);
+                System.out.println("Now you have " + indexOfTask + " tasks in the list.");
+                printLine();
+            }
+            else{
```

Comment:
There is a coding violation here as the else should be just after the curly bracket.